src/Rector/Deprecation/RemoveDeprecationHelperRector.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Rector\AbstractDrupalCoreRector;
+use DrupalRector\Rector\ValueObject\RemoveDeprecationHelperConfiguration;
+use PhpParser\Node;
+use PhpParser\NodeDumper;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class RemoveDeprecationHelperRector extends AbstractRector implements ConfigurableRectorInterface
+{
+
+    /**
+     * @var array|RemoveDeprecationHelperConfiguration[]
+     */
+    private array $configuration;
+
+    /**
+     * @inheritdoc
+     */
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Fixes deprecated user_password() calls',[
+            new CodeSample(
+                <<<'CODE_BEFORE'
+$pass = user_password();
+$shorter_pass = user_password(8);
+CODE_BEFORE
+                ,
+                <<<'CODE_AFTER'
+$pass = \Drupal::service('password_generator')->generate();
+$shorter_pass = \Drupal::service('password_generator')->generate(8);
+CODE_AFTER
+            )
+        ]);
```

Comment:
I don't think this matches. It's a Rector to Rector our replacements of `backwardsCompatibleCall`?

Reply:
Or is this a Rector to revert BC calls when they're no longer needed?

Reply:
Yes this rector is to remove BC calls when you no longer need those. What if we are at Drupal 11 -> 12 cycle, then we need to probably remove Drupal 10 -> 11 BC calls at some point.

Also if a new major version of your module is released this could help you remove all the BC calls easily if you want to drop a major version

Reply:
But you are also right. This example is not correct and need to be updated with a depracatedhelper call to a non-deprecated helper call. Oops.

Reply:
I wil fix this while rebasing and using a true Drupal 10 deprecataion

Reply:
Resolved